### ISSUE TITLE
[🐛 Bug]: Tests using describe.skip are marked as failed in Browserstack

### ISSUE THREAD

**Issue by alexradulescuqa** [webdriverio/webdriverio#12337]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.29.3

### Node.js Version

20.11.0

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
{
      platformName: 'Android',
      acceptInsecureCerts: true,
      'appium:automationName': 'UiAutomator2',
      'appium:autoGrantPermissions': true,
      'appium:newCommandTimeout': 240,
      'bstack:options': {
        deviceName: 'Samsung Galaxy S23',
        osVersion: '13.0',
        platformName: 'android',
        debug: true,
        networkLogs: true,
        appiumVersion: '2.0.1',
      },
    },
```

### What happened?

I'm using the mocha test framework. `describe.skip` tests are being marked as Failed on Browserstack.
The test is basically started on their end, but it's marked as failed instead of skipped on Browserstack.

In the meantime, the terminal shows the tests as being skipped.

I'm using these related packages:
"@wdio/browserstack-service": "^8.29.3",
"@wdio/appium-service": "^8.29.3",
"@wdio/local-runner": "^8.29.3",
"@wdio/mocha-framework": "^8.29.3",

### What is your expected behavior?

The test skipped with describe.skip will not start at all on Browserstack and will be marked as skipped there.

### How to reproduce the bug.

Do any skipped mocha test at the describe level. Use `describe.skip(() => { expect(true).toBe(true) })` and this should be marked as failed in Browserstack.

### Relevant log output

```typescript
no relevant log output
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by jemishgopani** [comment]
@alexradulescuqa can you please attach reproducible example here?

**Comment by alexradulescuqa** [comment]
Hey @jemishgopani . Sure. I will do it in a few days. I'm a little caught up with work at the moment.

**Comment by alexradulescuqa** [comment]
I looked into it some more. If I am not mistaken, this is happening due to how Mocha does things.
However, Browserstack marks the tests as failed instead of pending.
Mocha docs say that they should be pending. :thinking: 

Also, according to documentation:
> Note: Code in skipped suites, that is placed outside of hooks or tests is still executed, as mocha will still invoke the suite function to build up the suite structure for visualization.
This basically means that we'll always spin up the device in Browserstack even though the tests are marked as skipped, when using Mocha, instead of not starting the test at all.

https://mochajs.org/#inclusive-tests

I think that this is an expected behavior rather than a bug as a whole. Perhaps an improvement Browserstack can make, but not an issue on WebDriver.io's side.

Closing because of reasons above.
